Last-Update: 2012-12-23
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Forwarded: not-needed
Origin: http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/sys-block/seekwatcher/files/seekwatcher-0.12_p20091015-dash-fix.patch?view=markup
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Thu Feb 23 22:28:28 2012 +0300
Description: fix failure when /bin/sh points to dash.
 This patch fixes the following error:
    $ seekwatcher -t trace.blktrace -o output.png
    using tracefile ././trace
    sh: 1: Syntax error: Bad fd number
    Empty blktrace run found, exiting

--- a/cmd/seekwatcher
+++ b/cmd/seekwatcher
@@ -609,9 +609,9 @@
     for x in tracefiles:
         print "using tracefile %s" % os.path.join(trace_dir, x)
         fh = tempfile.NamedTemporaryFile(dir=".")
         os.system('blkparse -q -D ' + trace_dir + ' -i ' + x +
-                ' -d ' + fh.name + ' -O >& /dev/null')
+                ' -d ' + fh.name + ' -O > /dev/null 2>&1')
         loaddata(fh, rd)
     return rd
 
 def getlabel(i):
